syntax = "proto3";
package service_manage;





/*
CreateServiceTopologyTask请求
*/
message CreateServiceTopologyTaskRequest {
message Switch {
    /*
    是否允许将服务节点绑定至所归属的服务实例
    */
    bool connectNode2Service = 1 ;
    /*
    是否允许将服务节点从所归属的服务实例解绑
    */
    bool disconnectNode2Service = 2 ;
    /*
    是否允许根据服务节点间的关系新增服务间关系
    */
    bool connectService2Service = 3 ;
    /*
    是否允许根据服务节点间的关系解除服务间关系
    */
    bool disconnectService2Service = 4 ;
}
        
    
    /*
    执行命令（check：检查任务执行状态， execute：立即执行任务）
    */
    string command = 1 ;
    /*
    任务开关选项
    */
    Switch switch = 2 ;
}

/*
CreateServiceTopologyTask返回
*/
message CreateServiceTopologyTaskResponse {
    /*
    任务名称
    */
    string task = 1 ;
    /*
    任务运行状态
    */
    string status = 2 ;
}

/*
CreateServiceTopologyTaskApi返回
*/
message CreateServiceTopologyTaskResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    service_manage.CreateServiceTopologyTaskResponse data = 4 ;
}

